enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,600 @@
+delete the service after a subnet update. While this annotation was initially introduced to enable automatic scope
+changes in [Ingress Mutable Publishing Scope](/enhancements/ingress/mutable-publishing-scope.md), we've chosen to extend
+its usage to support the `Subnets` fields. The auto-delete annotation is not intended for end-users to use directly, but
+instead for configuration management tooling.
+
+### Implementation Details/Notes/Constraints
+
+When an IngressController is created with
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+specified, the Ingress Operator will set the
+`service.beta.kubernetes.io/aws-load-balancer-subnets` annotation
+on the LoadBalancer-type Service.
+
+However, if `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+is updated on an existing IngressController, the Ingress Operator will NOT
+update `service.beta.kubernetes.io/aws-load-balancer-subnets` until the
+cluster admin deletes the LoadBalancer-type service as instructed by the
+`LoadBalancerProgressing` status (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+When the Ingress Operator recreates the LoadBalancer-type service, it will
+then configure `service.beta.kubernetes.io/aws-load-balancer-subnets` with
+the new `Subnets` value.
+
+The only exception is if the `ingress.operator.openshift.io/auto-delete-load-balancer`
+annotation is set on the IngressController, in which case the operator automatically deletes
+the service and effectuates the subnet update.
+
+The `status.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+will eventually reflect the configured subnet value by mirroring the value of
+`service.beta.kubernetes.io/aws-load-balancer-subnets` on the service. There are
+two scenarios in which the `Subnets` status won't be equal to the `Subnets` spec:
+
+1. The cluster admin manually configured the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+   annotation on the service.
+2. The cluster admin configured the IngressController's `Subnets` spec, but hasn't
+   effectuated the change by deleting the service (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+
+This proposal's use of `status.endpointPublishingStrategy` is consistent with the approach in
+[LoadBalancer Allowed Source Ranges](/enhancements/ingress/lb-allowed-source-ranges.md),
+but diverges with the approach in [Ingress Mutable Publishing Scope](/enhancements/ingress/mutable-publishing-scope.md).
+The Ingress Mutable Publishing Scope design sets `status.endpointPublishingStrategy.loadBalancer.scope`
+equal to `spec.endpointPublishingStrategy.loadBalancer.scope`, which may not always reflect the actual scope of the
+load balancer. This proposal for `Subnets` ensures `status.endpointPublishingStrategy` reflects the _actual_ value
+(in our case, the annotation value).
```

Comment:
If this is true?
```suggestion
(in our case, the value of `service.beta.kubernetes.io/aws-load-balancer-subnets` on the Service).
```
If that's not true, I still don't understand what the _actual_ value is here.

Reply:
You are correct, that's the annotation I am talking about. Updated.